iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,320 @@
+{
+    m_sendBuffer << mqMessageTypeToString(MqMessageType::IMPL_INTERFACE) << std::string("/sender")
+                 << static_cast<uint32_t>(iox::capro::Interfaces::INTERNAL) << m_invalidRunnableName;
+
+    const auto status = m_senderRuntime->sendRequestToRouDi(m_sendBuffer, m_receiveBuffer);
+
+    EXPECT_EQ(false, status);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSender_ReturnValue)
+{
+    const auto senderPort = m_senderRuntime->getMiddlewareSender(
+        iox::capro::ServiceDescription(99u, 1u, 20u), m_runnableName, iox::runtime::PortConfigInfo(11u, 22u, 33u));
+
+    EXPECT_EQ(iox::capro::ServiceDescription(99u, 1u, 20u), senderPort->m_serviceDescription);
+    EXPECT_EQ(22u, senderPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(33u, senderPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSender_DefaultArgs)
+{
+    const auto senderPort = m_senderRuntime->getMiddlewareSender(iox::capro::ServiceDescription(99u, 1u, 20u));
+
+    EXPECT_EQ(0u, senderPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(0u, senderPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSender_SenderlistOverflow)
+{
+    m_errorHandlerCalled = false;
+
+    for (uint32_t i = 0u; i < iox::MAX_PORT_NUMBER; ++i)
+    {
+        m_senderRuntime->getMiddlewareSender(iox::capro::ServiceDescription(i, i + 1u, i + 2u));
+    }
+
+    EXPECT_TRUE(m_errorHandlerCalled);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareReceiver_ReturnValue)
+{
+    auto receiverPort = m_receiverRuntime->getMiddlewareReceiver(
+        iox::capro::ServiceDescription(99u, 1u, 20u), m_runnableName, iox::runtime::PortConfigInfo(11u, 22u, 33u));
+
+    EXPECT_EQ(iox::capro::ServiceDescription(99u, 1u, 20u), receiverPort->m_serviceDescription);
+    EXPECT_EQ(22u, receiverPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(33u, receiverPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareReceiver_DefaultArgs)
+{
+    auto receiverPort = m_receiverRuntime->getMiddlewareReceiver(iox::capro::ServiceDescription(99u, 1u, 20u));
+
+    EXPECT_EQ(0u, receiverPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(0u, receiverPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareReceiver_ReceiverlistOverflow)
+{
+    m_errorHandlerCalled = false;
+
+    for (uint32_t i = 0u; i < iox::MAX_PORT_NUMBER + 1; ++i)
+    {
+        m_senderRuntime->getMiddlewareReceiver(iox::capro::ServiceDescription(i, i + 1u, i + 2u));
+    }
+
+    EXPECT_TRUE(m_errorHandlerCalled);
+}
+
+
+TEST_F(PoshRuntime_test, GetServiceRegistryChangeCounter_ReturnValue)
+{
+    auto runnableData = m_senderRuntime->getServiceRegistryChangeCounter();
+
+    // Roudi internally  calls 5 servieces before application registers its service
+    EXPECT_EQ(5u, *runnableData);
+}
+
+
+TEST_F(PoshRuntime_test, GetServiceRegistryChangeCounter_OfferStopOfferService)
+{
+    auto initial_value = m_senderRuntime->getServiceRegistryChangeCounter();
```

Comment:
Please no snake_case unless it is a library call (e.g. STL) or STL-like code.

Reply:
updated